internal/internal_task_pollers.go

```diff
@@ -255,76 +254,32 @@ func (wtp *workflowTaskPoller) processWorkflowTask(workflowTask *workflowTask) e
 	// close doneCh so local activity worker won't get blocked forever when trying to send back result to laResultCh.
 	defer close(doneCh)
 
-process_WorkflowTask_Loop:
 	for {
+		var response *s.RespondDecisionTaskCompletedResponse
 		startTime := time.Now()
 		workflowTask.doneCh = doneCh
 		workflowTask.laResultCh = laResultCh
 		completedRequest, wc, err := wtp.taskHandler.ProcessWorkflowTask(workflowTask)
-		if err == nil && completedRequest == nil {
-			// decision task cannot complete because it is waiting for local activity to finish
-			// we need a timer to force complete it to avoid the decision task timeout on server.
-		wait_LocalActivity_Loop:
-			for {
-				deadlineToTrigger := time.Duration(float32(ratioToForceCompleteDecisionTaskComplete) * float32(wc.GetDecisionTimeout()))
-				delayDuration := startTime.Add(deadlineToTrigger).Sub(time.Now())
-				select {
-				case <-time.After(delayDuration):
-					// force complete
-					response, err := wtp.forceRespondDecisionTaskCompleted(wc, workflowTask, startTime)
-					if err != nil {
-						return err
-					}
-					if response == nil || response.DecisionTask == nil {
-						return nil
-					}
-
-					// we are getting new decision task, so reset the workflowTask and continue process the new one
-					workflowTask = wtp.toWorkflowTask(response.DecisionTask)
-					continue process_WorkflowTask_Loop
-
-				case lar := <-laResultCh:
-					// local activity result ready
-					completedRequest, err := wtp.processLocalActivityResult(lar.task.workflowTask, lar)
-
-					if _, ok := err.(*workflowContextAlreadyDestroyedError); ok {
-						return nil
-					}
-
-					if err == nil && completedRequest == nil {
-						// decision task is not done yet, still waiting for more local activities
-						continue wait_LocalActivity_Loop
-					}
-
-					response, err := wtp.RespondTaskCompletedWithMetrics(completedRequest, err, workflowTask.task, startTime)
-					if err != nil {
-						return err
-					}
-					if response == nil || response.DecisionTask == nil {
-						return nil
-					}
-
-					// we are getting new decision task, so reset the workflowTask and continue process the new one
-					workflowTask = wtp.toWorkflowTask(response.DecisionTask)
-					continue process_WorkflowTask_Loop
-				}
+		if _, ok := err.(*localActivityTimedOutError); ok {
+			// force complete
+			response, err = wtp.forceRespondDecisionTaskCompleted(wc, workflowTask, startTime)
```

Comment:
I might be missing something, but I don't see how after this change the "heartbeating" of the workflow task is going to work.